### PR TITLE
Refactor api

### PR DESCRIPTION
**Pull request by GlennLz**

### CODE REVIEW

**Comment by DrunkenToast** [comment]
I don't know if this is finished yet but people depend on this so we might want to speed up this PR.

**Comment by GlennLz** [comment]
I haven't been able to reach endri yet to discuss, but you may expect this to be done by tonight or tomorrow night.

**Comment by Dries1234** [comment]
Use .env for things like DB config, also when making the controllers for tables make sure you don't forget any :)

File: tsconfig.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
       "strictNullChecks": true,              /* Enable strict null checks. */
       // "strictFunctionTypes": true,           /* Enable strict checking of function types. */
       // "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
-      // "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
+       "strictPropertyInitialization": false,  /* Enable strict checking of property initialization in classes. */
```
Comment (Dries1234) [inline tsconfig.json:31]: Please keep this on true, this is the entire point of typescript. It's just bad practice to disable this.

Comment (GlennLz) [reply tsconfig.json:31]: I was not aware this was changed. Will do!

**Comment by GlennLz** [comment]
build fails because of tests but works locally

**Review by EndriFucia — APPROVED** [review]

**Pull request merged**